wrappers/models_configurations/base_config.py

```diff
@@ -0,0 +1,24 @@
+
+from typing import Any
+from pathlib import Path
+from .config_types import ConfigTypes
+import json
+from abc import ABC, abstractmethod
+
+class BaseConfiguration(ABC):
+    def __init__(self, api_key: str) -> None:
+        self.api_key = api_key
+
+    def to_dict(self) -> dict[str, Any]:
+        return self.__dict__
+    
+    def to_file(self) -> Path:
+        config_path = Path(__file__).parent / self._get_config_file_name()
+        with open(config_path, 'w') as f:
+            f.write(json.dumps(self.to_dict()))
+
+        return config_path
```

Comment:
save_configuration_to_file (or better) is clearer in intent, as to_file makes it sounds as a function converting an object to a File and returning it. returning the path in this context is meaningless.

Reply:
I think that the name of the function shouldn't repeat the class context, I'll go with `save_to_file `

```python 
config = BaseConfiguration(...)

config.save_configuration_to_file()  # repeating the context of configuration

config.save_to_file() # clearer IMO
```